pkg/client/client.go

```diff
@@ -315,12 +315,20 @@ type imageFactory struct {
 	keychain     authn.Keychain
 }
 
-func (f *imageFactory) NewImage(repoName string, daemon bool, imageOS string) (imgutil.Image, error) {
-	platform := imgutil.Platform{OS: imageOS}
+func (f *imageFactory) NewImage(repoName string, daemon bool, target dist.Target) (imgutil.Image, error) {
+	platform := imgutil.Platform{OS: target.OS, Architecture: target.Arch}
 
+	// daemon only uses os/arch
 	if daemon {
 		return local.NewImage(repoName, f.dockerClient, local.WithDefaultPlatform(platform))
 	}
 
+	// when targeting a registry, we need to use variant and osVersion to hit the correct image
+	platform.Variant = target.ArchVariant
+	if len(target.Distributions) > 0 {
+		// We assume the given target's distributions were already expanded, we should be dealing with just 1 distribution name and version.
+		platform.OSVersion = target.Distributions[0].Version
+	}
```

Comment:
Do we actually need this part?

Reply:
In this case, We need it  for setting the metadata into the Config file I think

Reply:
Could we change the comment above then? Because that talks about fetching the image (hence my confusion)